### PR TITLE
Cleanup of the gRPC API for AxonServer

### PR DESCRIPTION
**Pull request by abuijze**
- Added documentation to all messages and rpc requests
- Changed some of the names of types and properties to better reflect their purpose
- All changes are binary compatible with their previous format

### CODE REVIEW

File: src/main/proto/common.proto [hunk 1]
```diff
@@ -2,50 +2,102 @@ syntax = "proto3";
...
     TIMEOUT = 2;
+
+    /* key indicating that the requester expects at most the given number of results from this message. Use -1 for unlimited. */
     NR_OF_RESULTS = 3;
-    ORIGINAL_MESSAGE_ID = 4;
-    TARGET_CLIENT = 5;
-    CONTEXT = 6;
+
+    // do not reuse reserved number to ensure forward and backward compatibility
+    reserved 4, 5, 6, 15;
```
Comment (MGathier) [inline src/main/proto/common.proto]: 15?

File: src/main/proto/control.proto [hunk 2]
```diff
@@ -2,95 +2,186 @@ syntax = "proto3";
 package io.axoniq.axonserver.grpc.control;
 option java_multiple_files = true;
 
+/* Service describing operations for connecting to the AxonServer platform.
+
+ Clients are expected to use this service on any of the Platform's Admin nodes to obtain connection information of the
+ node that is should set up the actual connection with. On that second node, the clients should open an instruction
```
Comment (smcvb) [inline src/main/proto/control.proto]: Typo: `...node that it should set...` i.o. `...node that is should set...`

File: src/main/proto/control.proto [hunk 3]
```diff
@@ -2,95 +2,186 @@ syntax = "proto3";
...
     int32 active_threads = 3;
+
+    /* Flag indicating whether the processor is running */
     bool running = 4;
+
+    /* Flag indicating whether the processor, when stopped, did so because of an irrecoverable Error */
     bool error = 5;
-    repeated EventTrackerInfo event_trackers_info = 6;
+
+    /* Status details of each of the Segment for which Events are being processed. This is only provided by Tracking
```
Comment (smcvb) [inline src/main/proto/control.proto]: Typo: `...of the Segments for which...` i.o. `...of the Segment for which...`

File: src/main/proto/control.proto [hunk 4]
```diff
@@ -2,95 +2,186 @@ syntax = "proto3";
...
+       Event Processors.
+     */
+    repeated SegmentStatus segment_status = 6;
+
+    /* The number of threads the processor has available to assign to Segments.
+       Will report 0 is all threads are assigned a Segment.
```
Comment (smcvb) [inline src/main/proto/control.proto]: Typo: `Will report 0 if all threads...` i.o. `Will report 0 is all threads...`

File: src/main/proto/event.proto [hunk 5]
```diff
@@ -35,118 +36,229 @@ service EventStore {
...
     string query = 1;
+
+    /* The number of results AxonServer may send before new permits need to be provided */
     int64 number_of_permits = 2;
+
+    /* Whether to keep the query running against incoming events once the Head of the Stream is reached */
     bool live_events = 3;
 }
 
+/* A message describing the a response to a Query request */
```
Comment (smcvb) [inline src/main/proto/event.proto]: Typo : `A message describing a response to a Query request` i.o. `A message describing the a response to a Query request`

File: src/main/proto/event.proto [hunk 6]
```diff
@@ -35,118 +36,229 @@ service EventStore {
...
     repeated QueryValue id_values = 1;
+
+    /* The sorting values to use when sorting this response compared to the others. */
     repeated QueryValue sort_values = 2;
+
+    /* The actual data values for each of the columns, as a column name -> value mapping */
     map<string, QueryValue> values = 3;
 }
 
+/* Describins the combination of an Aggregate Identifier and first expected Sequence number when opening an Aggregate-specific Event Stream */
```
Comment (smcvb) [inline src/main/proto/event.proto]: Typo: `Describes the combination...`  i.o. `Describins the combination...`

File: src/main/proto/query.proto [hunk 7]
```diff
@@ -3,113 +3,268 @@ package io.axoniq.axonserver.grpc.query;
...
     SerializedObject payload = 4;
+
+    /* Any Meta Data describing the context of the Response Message */
     map<string, MetaDataValue> meta_data = 5;
+
+    /* Any instructions for components Routing or Handling the Response Message */
     repeated ProcessingInstruction processing_instructions = 6;
+
+    /* The unique identifier of the Query to which this is a response */
     string request_identifier = 7;
```
Comment (MGathier) [inline src/main/proto/query.proto]: Add note that 15 is reserved here.

File: src/main/proto/event.proto [hunk 8]
```diff
@@ -35,118 +36,229 @@ service EventStore {
...
 message Event {
-    /* A message id */
+
+    /* The unique identifier of this event */
     string message_identifier = 1;
+
+    /* The identifier of the Aggregate instance that published this event, if any */
     string aggregate_identifier = 2;
+
+    /* The sequence number of the Event in the of the Aggregate instance that published it, if any */
```
Comment (smcvb) [inline src/main/proto/event.proto]: Typo: `...of the Event in the Aggregate instance...` i.o. `...of the Event in the of the Aggregate instance...`

File: src/main/proto/query.proto [hunk 9]
```diff
@@ -3,113 +3,268 @@ package io.axoniq.axonserver.grpc.query;
...
+        /* Grant permits to AxonServer to send a number of messages to the client */
         FlowControl flow_control = 3;
+
+        /* Sends a Response to a Query received via the inbound stream */
         QueryResponse query_response = 4;
+
+        /* Indicator that all responses for Query have been sent */
         QueryComplete query_complete = 5;
+
+        /* Sends a response for a Subscription Query that have been received via the inbound stream */
```
Comment (smcvb) [inline src/main/proto/query.proto]: Typo: `...that has been received...` i.o. `...that have been received...`

File: src/main/proto/query.proto [hunk 10]
```diff
@@ -3,113 +3,268 @@ package io.axoniq.axonserver.grpc.query;
...
 message QueryUpdateComplete {
+
+    /* The identifier of the Client instance providing the Update */
     string client_id = 2;
+
+    /* The Component Name of the Client providing the Update */
     string component_name = 3;
 }
 
+/* Message indicating that an Error occurred and that no Updates will sent for a Subscription Query */
```
Comment (smcvb) [inline src/main/proto/query.proto]: Typo: `...no Updates will be sent for a...` i.o. `...no Updates will sent for a...`

File: src/main/proto/query.proto [hunk 11]
```diff
@@ -3,113 +3,268 @@ package io.axoniq.axonserver.grpc.query;
...
 message SubscriptionQueryResponse {
+
+    /* The unique identifier for this message */
     string message_identifier = 1;
+
+    /* The identifier of the subscription query this is a response for */
     string subscription_identifier = 2;
+
+    /* The actual response. The `initial_result` message is sent as a response to `get_initial_result`. An `update`
+       messages are sent for each update available for the query, even before the Initial Result is supplied. The
```
Comment (smcvb) [inline src/main/proto/query.proto]: Typo: `...messages is sent for...` i.o. `...messages are sent for...`

**Pull request merged**